### PR TITLE
Documentation modules v2

### LINKED ISSUE #3023: Modules V2 - Docs
- When documenting Modules V2, add notes that these features are under development, that we can later remove. 
- [ ] Update modules.rst to explain both module formats 
- [ ] Update release.rst to explain that we have two ways of distributing, explain both
- [ ] Update `reference/projectyml.rst` to also show how this looks as 'yaml' (now it shows python only, may be hard to understand for some people)
- [ ] Update `reference/projectyml.rst` (by updating the `inmanta.module.ProjectMetadata`) to also be able to point to pypi indexes.

``` yaml
repos:
  - https://github.com/inmanta/
  - url: https://github.com/inmanta/
    type: git
  - url: https://pypi.org/simple/
    type: pypi
```

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

Added documentation modules v2.

closes #3023 

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation 

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Review by wouterdb — COMMENTED** [review]
(forgot to submit my review)

File: docs/model_developers/modules.rst [hunk 1]
```diff
@@ -2,26 +2,108 @@
...
+.. code-block:: ini
+
+    [metadata]
+    name = inmanta-module-mod1
+    version = 1.2.3
+    license = Apache 2.0
+
+    [options]
+    install_requires =
+      net ~=0.2.4
```
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply docs/model_developers/modules.rst:72]: Yes. You are right. This creates the strange side-effect that modules are pinned in the `setup.cfg` using their package name, while the `project.yml` uses the module name. Should we also support package names in the `project.yml` file?

[the remark this answers is not in the document]
Comment (wouterdb) [reply docs/model_developers/modules.rst]: I think this is ok

**Review by wouterdb — APPROVED** [review]

File: docs/model_developers/release.rst [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 =================
 
-Inmanta modules are versioned based on git tags. The current version is reflected in the ``module.yml`` file and in the
-commit is should be tagged in the git repository as well. To ease the use inmanta provides a command (``inmanta modules
-commit``) to modify module versions, commit to git and place the correct tag.
+.. warning::
+
+   The V2 module format is currently under development.
+
+Inmanta modules are versioned based on git tags. The current version is reflected in the ``module.yml`` file for V1 modules or
```
Comment (sanderr) [inline docs/model_developers/release.rst:8]: This is not necessarily the case for V2 modules, is it? I assume we will continue tagging modules but it isn't strictly part of the public requirements I think.

File: docs/model_developers/release.rst [hunk 3]
```diff
@@ -57,6 +61,23 @@ Or for the current project
...
+
+Distributing an Inmanta module using a Git repository happens by storing the source code of that module on a Git repository
+that is accessible by the Inmanta orchestrator. The orchestrator will clone the source code of the module and install it in the
+Inmanta project.
 
-For more information about the ``module.yml`` file see :ref:`module_yml`.
+Python package distribution format
+----------------------------------
 
+Modules defined in the V2 module format can be distributed as a Python package. Run the ``inmanta module package`` command in
```
Comment (sanderr) [inline docs/model_developers/release.rst:80]: I was going with `inmanta module build` in my PR, though I have no real preference (not a change request per se).

**Review by sanderr — CHANGES_REQUESTED** [review]

File: docs/model_developers/release.rst (hunk 2 above)
Comment (wouterdb) [reply docs/model_developers/release.rst:8]: I think it is still best practice, so I would keep it in the docs.

Comment (sanderr) [reply docs/model_developers/release.rst:8]: All right, I can get behind that.

**Review by sanderr — APPROVED** [review]

File: docs/model_developers/release.rst (hunk 3 above)
Comment (arnaudsjs) [reply docs/model_developers/release.rst:80]: I was doubting between those two as well. I don't have a real preference either. I think `build` is probably more in line with the Python terminology. I will update it.

**Pull request closed without merging**